### PR TITLE
Add 'Client.list_metrics' API wrapper.

### PR DESCRIPTION
**Pull request by tseaver**
Also, add 'Client.metric' factory.

~~Uses #1588 as a base.~~

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes Rebased after merging #1588 to the `logging-api` branch.  PTAL

File: gcloud/logging/client.py [hunk 1]
```diff
@@ -79,7 +80,8 @@ def _entry_from_resource(self, resource, loggers):
         raise ValueError('Cannot parse job resource')
 
     def list_entries(self, projects=None, filter_=None, order_by=None,
-                     page_size=None, page_token=None):
+                     page_size=None,
+                     page_token=None):
```
Comment (dhermes) [inline gcloud/logging/client.py]: Why this change?

Comment (tseaver) [inline gcloud/logging/client.py]: Dunno.  Might've come about while resolving conflicts with the `list_sinks` stuff?

Comment (tseaver) [inline gcloud/logging/client.py]: affbc5d

File: gcloud/logging/client.py [hunk 2]
```diff
@@ -112,6 +114,7 @@ def list_entries(self, projects=None, filter_=None, order_by=None,
                   more topics can be retrieved with another call (pass that
                   value as ``page_token``).
         """
+        # pylint: disable=too-many-branches
```
Comment (dhermes) [inline gcloud/logging/client.py]: Ditto here, why did it become necessary? Also where is the `enable=` statement?

File: gcloud/logging/client.py [hunk 3]
```diff
@@ -190,3 +192,56 @@ def list_sinks(self, page_size=None, page_token=None):
...
+
+        :type filter_: string
+        :param filter_: the advanced logs filter expression defining the
+                        entries tracked by the metric.
+
+        :type description: string
+        :param description: the description of the metric to be constructed.
+
+        :rtype: :class:`gcloud.pubsub.logger.Logger`
+        :returns: Logger created with the current client.
```
Comment (dhermes) [inline gcloud/logging/client.py]: Docstring copy-pasta?

File: gcloud/logging/client.py [hunk 4]
```diff
@@ -190,3 +192,56 @@ def list_sinks(self, page_size=None, page_token=None):
...
+
+        :type page_token: string
+        :param page_token: opaque marker for the next "page" of metrics. If not
+                           passed, the API will return the first page of
+                           metrics.
+
+        :rtype: tuple, (list, str)
+        :returns: list of :class:`gcloud.logging.metric.Metric`, plus a
+                  "next page token" string:  if not None, indicates that
+                  more metrics can be retrieved with another call (pass that
```
Comment (dhermes) [inline gcloud/logging/client.py]: Open parenthesis is never closed `(pass that`

File: gcloud/logging/client.py (hunk 2 above)
Comment (tseaver) [inline gcloud/logging/client.py]: > Ditto here, why did it become necessary?

At some point, I was suppressing an actual pylint failure here:  it would have been silly to split that function up.  Now, it appears to be unnecessary (maybe merged changes on `master`?)

> Also where is the enable= statement?

[Disabling comments are block-scoped](https://docs.pylint.org/faq.html#is-it-possible-to-locally-disable-a-particular-message).  Moot after 3101fb0.

File: gcloud/logging/client.py (hunk 3 above)
Comment (tseaver) [inline gcloud/logging/client.py]: 02eecd1

File: gcloud/logging/client.py (hunk 4 above)
Comment (tseaver) [inline gcloud/logging/client.py]: f1ca149

**Comment by tseaver** [comment]
@dhermes PTAL.

**Comment by dhermes** [comment]
LGTM
